TEveEventDisplay/src/TEveMu2eDataInterface.cc

```diff
@@ -290,6 +290,104 @@ namespace mu2e{
   }
 
 
+using LHPT = KinKal::PiecewiseTrajectory<KinKal::LoopHelix>;
+using CHPT = KinKal::PiecewiseTrajectory<KinKal::CentralHelix>;
+using KLPT = KinKal::PiecewiseTrajectory<KinKal::KinematicLine>;
+template<class KTRAJ> void TEveMu2eDataInterface::AddKinKalTrajectory( std::unique_ptr<KTRAJ> &trajectory, TEveMu2eCustomHelix *line, TEveMu2eCustomHelix *line_twoDXY, TEveMu2eCustomHelix *line_twoDXZ){
+  double t1=trajectory->range().begin();
+  double t2=trajectory->range().end();
+
+  double x1=trajectory->position3(t1).x();
+  double y1=trajectory->position3(t1).y();
+  double z1=trajectory->position3(t1).z();
+  GeomHandle<DetectorSystem> det;
+  XYZVectorF pos(x1,y1,z1);
+  CLHEP::Hep3Vector p = GenVector::Hep3Vec(pos);
+  CLHEP::Hep3Vector InMu2e = det->toMu2e(p);
+  line->SetPoint(0,pointmmTocm(InMu2e.x()), pointmmTocm(InMu2e.y()) , pointmmTocm(InMu2e.z()));
+  line_twoDXY->SetPoint(0,pointmmTocm(x1), pointmmTocm(y1) , pointmmTocm(z1));
+  line_twoDXZ->SetPoint(0,pointmmTocm(x1), pointmmTocm(y1) , pointmmTocm(z1));
+  for(double t=t1; t<=t2; t+=0.1)
+  {
+    const auto &p = trajectory->position3(t);
+    double xt=p.x();
+    double yt=p.y();
+    double zt=p.z();
+    XYZVectorF pos(x1,y1,z1);
+    CLHEP::Hep3Vector pmu2e = GenVector::Hep3Vec(pos);
+    CLHEP::Hep3Vector InMu2e = det->toMu2e(pmu2e);
+    line->SetNextPoint(pointmmTocm(InMu2e.x()), pointmmTocm(InMu2e.y()) , pointmmTocm(InMu2e.z()));
+    line_twoDXY->SetNextPoint(pointmmTocm(xt), pointmmTocm(yt), pointmmTocm(zt));
+    line_twoDXZ->SetNextPoint(pointmmTocm(xt), pointmmTocm(yt), pointmmTocm(zt));
+   // CLHEP::Hep3Vector p = GenVector::Hep3Vec(pos);
+    //          CLHEP::Hep3Vector InMu2e = det->toMu2e(p);
+         
+    
+  }
+}
+
+void TEveMu2eDataInterface::FillKinKalTrajectory(bool firstloop, std::tuple<std::vector<std::string>, std::vector<const KalSeedCollection*>> track_tuple, TEveMu2e2DProjection *tracker2Dproj, TEveProjectionManager *TXYMgr, TEveProjectionManager *TRZMgr, TEveScene *scene1, TEveScene *scene2){
+  std::cout<<"[REveMu2eDataInterface] AddKinKalTracks  "<<std::endl;
+  std::vector<const KalSeedCollection*> track_list = std::get<1>(track_tuple);
+  std::vector<std::string> names = std::get<0>(track_tuple);
+  std::vector<int> colour;
+  
+  for(unsigned int j=0; j< track_list.size(); j++){
+    const KalSeedCollection* seedcol = track_list[j];
+    colour.push_back(j+3);
+    if(seedcol!=0){  
+     for(unsigned int k = 0; k < seedcol->size(); k = k + 20){ 
```

Comment:
Why does k advance by 20?  it pretty much guarantees you'll only see the 1st track.

Reply:
set to k++